File: kinotic-frontend/apps/system/src/services/KinoticClusterInfoService.ts
import { Kinotic } from '@kinotic-ai/core'
import { SYSTEM_ZONE } from '@kinotic-ai/os-api'
import type { KinoticClusterInfo } from '@/domain/ClusterInfo'

/**
 * Queries the platform cluster state from the server's {@code KinoticClusterInfoService}
 * (published in the system zone). App-local proxy: the console is its only consumer.
 */
export class KinoticClusterInfoService {

    public getClusterInfo(): Promise<KinoticClusterInfo> {
        return Kinotic.serviceProxy(`${SYSTEM_ZONE}~org.kinotic.os.api.services.KinoticClusterInfoService`)
                      .invoke('getClusterInfo')
    }
}

export const CLUSTER_INFO_SERVICE = new KinoticClusterInfoService()

File: kinotic-management-api/src/main/java/org/kinotic/os/api/services/KinoticClusterInfoService.java
package org.kinotic.os.api.services;

import io.vertx.core.Future;
import org.kinotic.core.api.annotations.Publish;
import org.kinotic.core.api.annotations.Zone;
import org.kinotic.domain.api.model.cluster.KinoticClusterInfo;
import org.kinotic.domain.api.utils.DomainUtil;

/**
 * Provides information about the ignite Kinotic cluster.
 */
@Publish
@Zone(DomainUtil.SYSTEM_ZONE)
public interface KinoticClusterInfoService {
    
    /**
     * Returns the information about the ignite structures cluster.
     * 
     * @return the information about the ignite structures cluster
     */
    Future<KinoticClusterInfo> getClusterInfo();

}
